serving/src/main/java/ai/djl/serving/http/InferenceRequestHandler.java

```diff
@@ -468,24 +474,21 @@ void sendOutput(Output output, ChannelHandlerContext ctx) {
     }
 
     void onException(Throwable t, ChannelHandlerContext ctx) {
-        HttpResponseStatus status;
+        HttpResponseStatus status = httpResponseStatusMapper.getHttpStatusForException(t);
         if (t instanceof TranslateException || t instanceof BadRequestException) {
             logger.debug(t.getMessage(), t);
             SERVER_METRIC.info("{}", RESPONSE_4_XX);
-            status = HttpResponseStatus.BAD_REQUEST;
         } else if (t instanceof WlmException) {
```

Comment:
We might need to introduce a TimeoutException (extends EngineException) and implement this in PyProcess.
Currently if predict timeout, we throw EngineException. We might need to see how can we differentiate timeout vs normal TranslateException